neusomatic/include/msa.hpp

```diff
@@ -64,29 +44,40 @@ class GappedSeq {
   decltype(auto) ginv() const {return (ginv_);}
   int32_t left() const {return ginv_.left();}
   int32_t right() const {return ginv_.right();}
-  decltype(auto) bases() const {return (bases_);}
-  decltype(auto) gaps() const {return (gaps_);}
+  std::vector<Base> bases() const {return bases_;}
+  std::vector<int> gaps() const {return gaps_;}
 
   std::string to_string() const {
     std::string result;
     for (size_t i = 0; i < bases_.size(); ++i) {
       if (i != 0) {
-        for (int32_t j = 0; j < gaps_[i].len(); ++j) {
+        for (int32_t j = 0; j < gaps_[i]; ++j) {
           result += '-';
         }
       }
       result += bases_[i];
     }
     return result;
   }
+
+  int length() const {
+    int res = 0;
+    for (size_t i = 0; i < bases_.size(); ++i) {
```

Comment:
Do `bases_` and `gaps_` have the same size here? Can you not have `i` loop condition depend on `gaps_.size()`?

Reply:
that's @roger-liu-bina 's code.
@roger-liu-bina would you please comment?

Reply:
@marghoob Yes `bases_` and `gaps_` always have the same size.